keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlUnhappyVetoRepository.kt

```diff
@@ -31,14 +32,30 @@ class SqlUnhappyVetoRepository(
   private val sqlRetry: SqlRetry
 ) : UnhappyVetoRepository(clock) {
 
-  override fun markUnhappyForWaitingTime(resourceId: String, application: String, wait: Duration) {
+  override fun markUnhappyForWaitingTime(resourceId: String, application: String, wait: Duration?) {
     sqlRetry.withRetry(WRITE) {
       jooq.insertInto(UNHAPPY_VETO)
         .set(UNHAPPY_VETO.RESOURCE_ID, resourceId)
         .set(UNHAPPY_VETO.APPLICATION, application)
-        .set(UNHAPPY_VETO.RECHECK_TIME, calculateExpirationTime(wait).toTimestamp())
+        .run {
+          calculateExpirationTime(wait)?.toTimestamp().let { expiryTime ->
```

Comment:
I think it's a bit nicer if you use `?.let`:

```kotlin
calculateExpirationTime(wait)
  ?.toTimestamp()
  ?.let { expiryTime -> set(UNHAPPY_VETO.RECHECK_TIME, expiryTime) }
  ?: setNull(UNHAPPY_VETO.RECHECK_TIME)
```

Reply:
For some reason this behaves differently and breaks the tests. I will figure it out tho.